File: bignumber.c
#include <stdlib.h>
#include <stdio.h>

int main(){
	int count=0 ;
	int n ;
	printf("please enter a integer : ") ;
	scanf("%d" , &n) ;
	short a[200] ; // 先宣告一個矩陣(存取數字的矩陣)
	int m=0 , k , i , carry=0 ; // m 代表位數 ，k 是算階乘時的數字 ，i 是矩陣內每個位子的數值 ，carry 是進位的數字
	for (i=0 ; i<200 ; i++) //把每個位數都定為0
		a[i]=0 ;
	a[m]=1 ; //定個位數為1
	for (k=2 ; k<=n ; k++){
		for (i=0 ; i<=m ; i++){
			a[i]=a[i]*k ; //每一位乘k
			a[i]=a[i]+carry ;
			if((carry=a[i]/10)!=0 && i==m)
				m++ ; 
			a[i]=a[i]%10 ;
			count+=1 ;
		}
		count=0 ;
	}
	printf("%d!=" , n) ;
	for (i=m ; i>=0 ; i--){
		printf("%d" , a[i]) ;
	}
	printf("\n") ;
	system("pause") ;

}

File: bignumber.py
#儲存超大的數字(階乘)
n=5
#int(input("please enter a integer : "))
position=0 #位數
j=0 
carry=0 #進位的十位數
num=[] #存數字的矩陣
for i in range(10):#先預設一個矩陣，把每一位訂做0
	num.append(0) #先預設10個元素的矩陣
num[0]=1
for k in range(2,n+1):
	print("now is " , k) #現在要乘於k
#要乘到n
	for j in range(position+1) : #每個位數都乘一遍k
		num[j]=num[j]* k
		num[j]=num[j]+ carry
		print("old is " , num[j])
		carry=num[j]//10
		if (carry!=0) & (j==position)  : #如果在最高位乘的時候>10
			position+=1 #之後算的位數多1位
			num[position]=carry 
			carry=0 
		
		num[j]=num[j]%10
		print("j is " , j , "position is " , position)
		# print(carry)
		print("new is " , num[j] )
		print("carry is  " , carry, "\n")


		

print("%d ! = " %n)
#for i in range(len(num)) :
print(num)
print("\n")
